### PR TITLE
Fixing Creation of Opensuse data Files

### PR DESCRIPTION
**Pull request by Princee215**
As mentioned in #145 issue. All the OpenSUSE data files that are being generated have an empty JSON file, that was due to a buggy code that Arsh and I tried to correct. We use to have separate chunk of code for creation of different files, In this PR I've use **nested for loop** and try to reduce down the **repetition** and the **lines of code**.
Once this get a green flag, I'll subsequently update the data files in the data repository that are empty right now.

@pleia2 and @arshPratap please take a look and let me know if there's anything that needs to be done.

### CODE REVIEW

File: bin/package_build.py [hunk 1]
```diff
@@ -49,82 +49,39 @@ def debian():
...
+			try:
+				req = requests.get(src_url)
+				data_source = req.content
+				if req.status_code == 404:
+					raise Exception(f"404 Directory for Opensuse list not found")
+			except Exception as e:
+				print("Couldn't pull. Error: ",str(e))
+			else:
+				data = json.loads(data_source)
+				regex_pattern = r"-(.*?)-"
```
Comment (arshPratap) [inline bin/package_build.py:69]: @Princee215 could you move the `regex_pattern` above (its seems like a constant), ... define it just below `source_data` list

File: bin/package_build.py [hunk 2]
```diff
@@ -49,82 +49,39 @@ def debian():
...
+					reg_match = re.search(regex_pattern, name)
+					if reg_match:
+						package_name = name[reg_match.end():][::-1]
+						version = reg_match[0][1:-1][::-1]
+						if package_name == None or version == None:
+							continue
+						data_dict = {"packageName":package_name,"description":"","version":version}
+						opensuse_list.append(data_dict)
+		q_index = q[i]
+		file_name = f'{DATA_FILE_LOCATION}/{q_index}'
```
Comment (arshPratap) [inline bin/package_build.py:81]: rename `q_index` to `file_name` and rename the `file_name` to `file_path `

File: bin/package_build.py [hunk 3]
```diff
@@ -49,82 +49,39 @@ def debian():
...
+		with open(file_name,'w') as file:
+			json.dump(opensuse_list,file,indent=2)
+			print(f"Saved!\nfilename: {q_index}")
```
Comment (arshPratap) [inline bin/package_build.py:84]: remove `q_index` and rename as defined above

**Comment by arshPratap** [comment]
Thanks @Princee215 ..works really well..just some minor fixes thats all

**Comment by arshPratap** [comment]
@Princee215 lemme know when you have fixed the changes I have suggested ..  I would love to approve this PR
@pleia2 would love to hear your thoughts on this .
Thanks

**Comment by Princee215** [comment]
Sure, thanks @arshPratap, i see these changes. And I'll try to edit as soon as possible.

**Review by pleia2 — APPROVED** [review]

**Comment by arshPratap** [comment]
Great Work @Princee215

**Pull request merged**